### PR TITLE
wps execution limits

### PR DESCRIPTION
**Pull request by julian1**

### CODE REVIEW

**Comment by jonescc** [comment]
I'll review

File: src/extension/wps/wps-core/src/main/java/org/geoserver/wps/WPSInfo.java [hunk 1]
```diff
@@ -140,11 +153,24 @@
      */
     public abstract int getMaxSynchronousExecutionTime();
 
+     /**
+     * How many seconds a process can run or queue in asynchronous mode (with the user waiting on the HTTP
```
Comment (jonescc) [inline src/extension/wps/wps-core/src/main/java/org/geoserver/wps/WPSInfo.java]: Is asynchronous mode the right description here?

File: src/extension/wps/wps-core/src/main/java/org/geoserver/wps/WPSInfo.java [hunk 2]
```diff
@@ -140,11 +153,24 @@
...
+     * connection) before it gets killed by the WPS container (0 or a negative value means no limit)
+     */
+    public abstract int getMaxSynchronousTotalTime();
+
     /**
      * Sets how many seconds a process can run in asynchronous mode (with the user waiting on the
```
Comment (jonescc) [inline src/extension/wps/wps-core/src/main/java/org/geoserver/wps/WPSInfo.java]: And here

File: src/extension/wps/wps-core/src/main/java/org/geoserver/wps/WPSInfo.java [hunk 3]
```diff
@@ -140,11 +153,24 @@
...
      * HTTP connection) before it gets killed by the WPS container (0 or a negative value means no
      * limit)
      */
     public abstract void setMaxSynchronousExecutionTime(int maxSynchronousExecutionTime);
+ 
+    /**
+     * Sets how many seconds a process can run or queue in asynchronous mode (with the user waiting on the
```
Comment (jonescc) [inline src/extension/wps/wps-core/src/main/java/org/geoserver/wps/WPSInfo.java]: And here

**Comment by jonescc** [comment]
Perhaps some unit tests would be good.  Normally required by us and Geoserver developers for code submissions.  Are there unit tests for the existing execution timeouts?

**Comment by julian1** [comment]
There don't appear to be prexisting unit tests for the execution timeouts.

File: src/extension/wps/wps-core/src/main/java/org/geoserver/wps/WPSInfo.java (hunk 3 above)
Comment (julian1) [inline src/extension/wps/wps-core/src/main/java/org/geoserver/wps/WPSInfo.java]: fixed.

**Comment by julian1** [comment]
uggh still working on this.

**Comment by julian1** [comment]
I've addressed comments and rebased. Could somebody merge this?

**Comment by danfruehauf** [comment]
I'll have a look quite soon.

File: src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/MaxExecutionTimeListener.java [hunk 4]
```diff
@@ -16,14 +16,17 @@
...
     long maxExecutionTime;
+    long maxTotalTime;
+    long queuedTime;
+    long startTime;
 
-    public MaxExecutionTimeListener(ProgressListener progress, long maxExecutionTime) {
+    public MaxExecutionTimeListener(ProgressListener progress, long maxExecutionTime, long maxTotalTime) {
         super(progress);
-        this.startTime = System.currentTimeMillis();
+        this.maxTotalTime = maxTotalTime;
```
Comment (danfruehauf) [inline src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/MaxExecutionTimeListener.java]: Doesn't really bother me, but to be in upstream PR shape, I think you want to reorder the parameters here, i.e. have `maxExecutionTime` before `maxTotalTime`.

**Comment by danfruehauf** [comment]
Just one little comment. If you addressed @jonescc comments, it's probably GTG.

File: src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/MaxExecutionTimeListener.java (hunk 4 above)
Comment (julian1) [inline src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/MaxExecutionTimeListener.java]: I've reordered the parameters to be consistent with other usage.

**Comment by danfruehauf** [comment]
GL!

**Comment by danfruehauf** [comment]
IMO that should make it in quite easily.

**Pull request merged**